sqlmesh/core/snapshot/definition.py

```diff
@@ -1482,12 +1482,17 @@ def missing_intervals(
             snapshot.intervals = snapshot.intervals.copy()
             snapshot.remove_interval(interval, execution_time)
 
+        missing_interval_end_date = snapshot_end_date
```

Comment:
```suggestion
        snapshot_end_date = min(to_datetime(snapshot_end_date), to_datetime(snapshot.node.end or snapshot_end_date)
```

Reply:
My initial implementation did this, however I observed that passing datetime objects instead of the strings to `snapshot.missing_intervals` caused "off by 1 interval" issues in the existing test suite.

So I did the check and then passed the string value, rather than passing the datetime value

Reply:
ah tricky one. yes, dates are inclusives and date times are exclusive